common/fixation_entry.go

```diff
@@ -996,37 +996,61 @@ func (fs *FixationStore) setVersion(ctx sdk.Context, val uint64) {
 	store.Set(types.KeyPrefix(types.FixationVersionKey), b)
 }
 
-func (fs *FixationStore) Export(ctx sdk.Context) []types.RawMessage {
-	store := prefix.NewStore(
-		ctx.KVStore(fs.storeKey),
-		types.KeyPrefix(fs.prefix))
-	iterator := sdk.KVStorePrefixIterator(store, []byte{})
-	defer iterator.Close()
+func (fs *FixationStore) Export(ctx sdk.Context) types.GenesisState {
+	gs := types.GenesisState{}
 
-	data := []types.RawMessage{}
-	for ; iterator.Valid(); iterator.Next() {
-		data = append(data, types.RawMessage{Key: iterator.Key(), Value: iterator.Value()})
+	gs.Version = fs.getVersion(ctx)
+
+	for _, EntryIndex := range fs.GetAllEntryIndices(ctx) {
```

Comment:
`GetAllEntryIndices()` skips entries marked deleted (but still in stale period or otherwise held as marker - see details in `deleteStaleEntries()`. You need to use `AllEntryIndicesFilter()` with a nil filter instead.

I suggest you add a test - generate some state with adds, deletes, refcount, future entries, stale, etc. Then do export than init to a second store, then compare the stores using the loop of the previous version of the export.  (That would have caught it).

Reply:
done https://github.com/lavanet/lava/pull/678/commits/6ee8bf73cd4a8c56bc341c44c26513105b9f97ec